hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -610,21 +639,255 @@ def set_net_pay(self):
+			self.deduct_tax_for_unsubmitted_tax_exemption_proof = 1
+			self.deduct_tax_for_unclaimed_employee_benefits = 1
+
+		# Get taxable unclaimed benefits
+		self.unclaimed_taxable_benefits = 0
+		if self.deduct_tax_for_unclaimed_employee_benefits:
+			self.unclaimed_taxable_benefits = self.calculate_unclaimed_taxable_benefits()
+			self.unclaimed_taxable_benefits += self.current_taxable_earnings_for_payment_days.flexi_benefits
+
+		# Total exemption amount based on tax exemption declaration
+		self.total_exemption_amount = self.get_total_exemption_amount()
+
+		# Employee Other Incomes
+		self.other_incomes = self.get_income_form_other_sources() or 0.0
+
+		# Total taxable earnings including additional and other incomes
+		self.total_taxable_earnings = (
+			self.previous_taxable_earnings
+			+ self.current_structured_taxable_earnings
+			+ self.future_structured_taxable_earnings
+			+ self.current_additional_earnings
+			+ self.other_incomes
+			+ self.unclaimed_taxable_benefits
+			- self.total_exemption_amount
+		)
+
+		# Total taxable earnings without additional earnings with full tax
+		self.total_taxable_earnings_without_full_tax_addl_components = (
+			self.total_taxable_earnings - self.current_additional_earnings_with_full_tax
+		)
+
+	def compute_income_tax_breakup(self):
+		if not self.payroll_period:
+			return
+
+		self.non_taxable_earnings = self.compute_non_taxable_earnings()
+
+		self.ctc = self.compute_ctc()
+
+		self.income_from_other_sources = self.get_income_form_other_sources()
+
+		self.total_earnings = self.ctc + self.income_from_other_sources
+
+		self.deductions_before_tax_calculation = self.compute_annual_deductions_before_tax_calculation()
+
+		self.standard_tax_exemption_amount = (
+			self.tax_slab.standard_tax_exemption_amount if self.tax_slab.allow_tax_exemption else 0.0
+		)
+
+		self.tax_exemption_declaration = (
+			self.get_total_exemption_amount() - self.standard_tax_exemption_amount
+		)
+
+		self.annual_taxable_amount = self.total_earnings - (
+			self.non_taxable_earnings
+			+ self.deductions_before_tax_calculation
+			+ self.tax_exemption_declaration
+			+ self.standard_tax_exemption_amount
+		)
+
+		self.income_tax_deducted_till_date = self.get_income_tax_deducted_till_date()
+
+		if hasattr(self, "total_structured_tax_amount"):
+			self.future_income_tax_deductions = (
+				self.total_structured_tax_amount - self.income_tax_deducted_till_date
+			)
+
+	def compute_ctc(self):
+		if hasattr(self, "previous_taxable_earnings"):
+			return (
+				self.previous_taxable_earnings
+				+ self.current_structured_taxable_earnings
+				+ self.future_structured_taxable_earnings
+				+ self.current_additional_earnings
+				+ self.other_incomes
+				+ self.unclaimed_taxable_benefits
+				+ self.non_taxable_earnings
+			)
+
+		else:
+			return 0.0
+
+	def compute_non_taxable_earnings(self):
+		non_taxable_earnings = 0.0
+		non_taxable_additional_salary = 0.0
+
+		prev_period_non_taxable_earnings = 0.0
```

Comment:
IG, these variables don't need to be initialized in advance since they aren't being used in any operations. They are just being assigned values by calling functions. So the function being called should take care of returning 0 instead of giving this responsibility to the place from where it's being called?

```suggestion
```